tools/snp.sh

```diff
@@ -164,6 +163,72 @@ verify_snp_host() {
   fi
 }
 
+verify_platform_snp_bit_status() {
+  if [ "$1" == "host" ]; then
+    # Get the host cpuid eax
+    local host_cpuid_eax
+    host_cpuid_eax=$(get_cpuid 0x8000001f eax)
+
+    # Map all the computed host sev/snp bit values in a single associative array
+    declare -A actual_sev_snp_bit_status=(
+      [SME]=$(( (${host_cpuid_eax} >> 0) & 1))
+      [SEV]=$(( (${host_cpuid_eax} >> 1) & 1))
+      [SEV-ES]=$(( (${host_cpuid_eax} >> 3) & 1))
+      [SNP]=$(( (${host_cpuid_eax} >> 4) & 1))
+    )
+
```

Comment:
Additionally to the CPUID (which checks if the feature is supported by the CPU) there is also an MSR that you can use to check if SME and SNP are enabled.

MSR 0xC0010010 bit 23 checks SME
MSR 0xC0010010 bit 24 checks SNP